[dynamo][misc] Replace UserFunctionVariable with VariableTracker build

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #165707
* #166251

Audit: To prevent future issues with functools.partial or callable
objects.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

anijain2305 committed:
[dynamo][misc] Replace UserFunctionVariable with VariableTracker build

Audit: To prevent future issues with functools.partial or callable
objects.

[ghstack-poisoned]

Lucaskabela reviewed (approved):
Thanks for propagating this through the codebase!

anijain2305 committed:
Update on "[dynamo][misc] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable
objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][misc] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable
objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge

clee2000 commented:
@pytorchbot revert -m "failing internal tests D85160820" -c ghfirst

anijain2305 committed:
Update on "[dynamo][misc] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable
objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]
